tests/unit/via/views/view_video_test.py

```diff
@@ -1,36 +1,65 @@
 from unittest.mock import sentinel
 
 import pytest
+from marshmallow.exceptions import ValidationError as MarshmallowValidationError
 from pyramid.httpexceptions import HTTPUnauthorized
 
+from via.exceptions import BadURL
 from via.views.view_video import view_video
 
+# webargs's kwargs injection into view functions falsely triggers Pylint's
+# no-value-for-parameter all the time so just disable it file-wide.
+# pylint: disable=no-value-for-parameter
+
 
 @pytest.mark.usefixtures("youtube_service")
 class TestViewVideo:
-    def test_it(self, pyramid_request, Configuration):
-        pyramid_request.matchdict["id"] = "abcdef"
+    def test_it(self, pyramid_request, Configuration, youtube_service, video_url):
+        youtube_service.get_video_id.return_value = sentinel.youtube_video_id
 
         response = view_video(pyramid_request)
 
-        assert response["client_embed_url"] == "http://hypothes.is/embed.js"
-        assert (
-            response["client_config"]
-            == Configuration.extract_from_params.return_value[1]
-        )
-        assert response["transcript"] == {
-            "segments": [
-                {
-                    "time": 0,
-                    "text": "First segment of transcript",
-                },
-                {
-                    "time": 30,
-                    "text": "Second segment of transcript",
-                },
-            ],
+        youtube_service.get_video_id.assert_called_once_with(video_url)
+        assert response == {
+            "client_embed_url": "http://hypothes.is/embed.js",
+            "client_config": Configuration.extract_from_params.return_value[1],
+            "transcript": {
+                "segments": [
+                    {
+                        "time": 0,
+                        "text": "First segment of transcript",
+                    },
+                    {
+                        "time": 30,
+                        "text": "Second segment of transcript",
+                    },
+                ],
+            },
+            "video_id": sentinel.youtube_video_id,
```

Comment:
I combined the several separate `assert`'s here into one but the only real difference is that `"video_id"` is now `sentinel.youtube_video_id`